### PR TITLE
#2 Place a marble on the board

### PR DESCRIPTION
**Pull request by Clebiez**
# (0.25) As Brice, I can place a marble on the board

- [x] Adding user input to place a marble
- [x] Check if user input is correct
- [x] Place marble in case of correct input
- [x] If board is full, stop the game

Peek 2021-01-25 17-17

### CODE REVIEW

File: src/board.py [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
+
+def get_position_if_valid(board, position):  
+    try:
+        if len(position) != 2:
+            return None
+
+        x = int(position[1]) - 1
+
+        # See https://stackoverflow.com/questions/5927149/get-character-position-in-alphabet
+        y = ord(position[0].lower()) - 97
```
Comment (fzaninotto) [inline src/board.py]: that's cryptic, and I don't want to go to StackOveflow to understand. Would you mind explaining here?

File: src/board.py [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
+    return [[0]*6 for _ in range(6)]
+
+def is_board_full(board):
+    for x, line in enumerate(board, 0):
+        for y, value in enumerate(line, 0):
+            if value == 0:
+                return False
+    return True
+
+def get_position_if_valid(board, position):
```
Comment (fzaninotto) [inline src/board.py]: I don't understand the code of this method. Perhaps you could detail the input format and add comments?

File: src/test_board.py [hunk 3]
```diff
@@ -1,16 +1,114 @@
...
+    
+    def test_is_board_full_should_return_true(self):
+        board = [[1]*6 for _ in range(6)]
+        print_board(board)
+
+        result = is_board_full(board)
+        
+        self.assertTrue(result)
+    
+    def test_get_position_if_valid_should_return_array_position(self):
```
Comment (fzaninotto) [inline src/test_board.py]: A good practice is to have 1 assertion per test. I suggest you use test providers (I don't know how they are called in Python, but it's a set of data you run your test on)

File: src/test_board.py [hunk 4]
```diff
@@ -1,16 +1,114 @@
...
+        board = [[0]*6 for _ in range(6)]
+        print_board(board)
```
Comment (fzaninotto) [inline src/test_board.py]: only in verbose mode

File: src/board.py [hunk 5]
```diff
@@ -1,12 +1,49 @@
...
+        y = ord(position[0].lower()) - 97
+
+        r = range(0, 6)
+
+        # If position is outside boundaries or on an already filled position.
+        if  x not in r or y not in r or board[x][y] != 0:
+            return None
+
+        # Return 2 dim array relative position (e.g [0, 0])
+        return [x, y]
```
Comment (HALLERPierre) [inline src/board.py]: I think you should use a tuple for that because it's fixed size and it should be immutable

File: src/test_board.py [hunk 6]
```diff
@@ -1,16 +1,114 @@
...
+
+        self.assertEqual(result, expected_result)
+
+    def test_add_marble_to_board(self):
+
+        board = [[0]*6 for _ in range(6)]
+        
+        expected_board = [[0]*6 for _ in range(6)]
+        
+        expected_board[0][0] = 1
```
Comment (fzaninotto) [inline src/test_board.py]: a bit weak, you test only the easiest position

File: src/board.py [hunk 7]
```diff
@@ -1,12 +1,49 @@
...
+    try:
+        if len(position) != 2:
+            return None
+
+        x = int(position[1]) - 1
+
+        # See https://stackoverflow.com/questions/5927149/get-character-position-in-alphabet
+        y = ord(position[0].lower()) - 97
+
+        r = range(0, 6)
```
Comment (HALLERPierre) [inline src/board.py]: Why 6 ? If it's the size of the board I think you should use at least a named const so you can change it easily if needed or asked

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/board.py [hunk 8]
```diff
@@ -1,12 +1,49 @@
...
+        if  x not in r or y not in r or board[x][y] != 0:
+            return None
+
+        # Return 2 dim array relative position (e.g [0, 0])
+        return [x, y]
+    except:
+        return None
+
+def add_marble_to_board(board, position):
+    board[position[0]][position[1]] = 1
```
Comment (HALLERPierre) [inline src/board.py:66]: Careful board is muted here, maybe deep clone it before that

**Review by fzaninotto — APPROVED** [review]

File: src/constant.py [hunk 9]
```diff
@@ -0,0 +1 @@
+ARRAY_SIZE = 6
```
Comment (mchaffotte) [inline src/constant.py:1]: array is too generic imo -> board_size ?

**Review by mchaffotte — APPROVED** [review]

**Pull request merged**